galsim/compound.py

```diff
@@ -902,3 +904,162 @@ def __setstate__(self, d):
 _galsim.SBFourierSqrt.__setstate__ = lambda self, state: 1
 _galsim.SBFourierSqrt.__repr__ = lambda self: \
         'galsim._galsim.SBFourierSqrt(%r, %r)'%self.__getinitargs__()
+
+
+class RandomWalk(Sum):
+    """
+
+    A class for generating a set of point sources distributed using a random
+    walk.  Uses of this profile include representing an "irregular" galaxy, or
+    adding this profile to an Exponential to represent knots of star formation.
+
+    Random walk profiles have a "shape noise" of g~0.05, but can be sheared to
+    give an additional ellipticity, for example to follow that of an associated
+    disk.
+    
+    Initialization
+    --------------
+    @param  npoints  Number of point sources to generate.
+    @param  hlr      Half light radius of the distribution of points.  This
+                     is the mean half light radius produced by an 
+                     infinite number of points.  A single instance will be
+                     noisy.
+    @param  flux     Total flux in all point sources.  Default 1
+    @param  nstep    Number of steps in random walk.  Default 40
```

Comment:
This doesn't really explain what nstep does.  Suggest something like "Number of random walk steps to take between each of the npoints."  Also, I don't have a feel for what are the pros and cons of increasing or decreasing this value, so perhaps a brief discussion in the intro about this would be warranted.